photon-core/src/main/java/org/photonvision/vision/target/TrackedTarget.java

```diff
@@ -128,6 +130,60 @@ public TrackedTarget(DetectionResult result, TargetCalculationParameters params)
         m_poseAmbiguity = result.getPoseAmbiguity();
     }
 
+    public TrackedTarget(ArucoDetectionResult result, TargetCalculationParameters params) {
+        m_targetOffsetPoint = new Point(result.getCenterX(), result.getCenterY());
+        m_robotOffsetPoint = new Point();
+
+        m_pitch =
+                TargetCalculations.calculatePitch(
+                        result.getCenterY(), params.cameraCenterPoint.y, params.verticalFocalLength);
+        m_yaw =
+                TargetCalculations.calculateYaw(
+                        result.getCenterX(), params.cameraCenterPoint.x, params.horizontalFocalLength);
+        var bestPose =
+                new Transform3d(result.getPose().getTranslation(), result.getPose().getRotation());
+
+        m_bestCameraToTarget3d = bestPose;
+
+        double[] xCorners = result.getxCorners();
+        double[] yCorners = result.getyCorners();
+
+        Point[] cornerPoints =
+                new Point[] {
+                    new Point(xCorners[0], yCorners[0]),
+                    new Point(xCorners[1], yCorners[1]),
+                    new Point(xCorners[2], yCorners[2]),
+                    new Point(xCorners[3], yCorners[3])
+                };
+        m_targetCorners = List.of(cornerPoints);
+        MatOfPoint contourMat = new MatOfPoint(cornerPoints);
+        m_approximateBoundingPolygon = new MatOfPoint2f(cornerPoints);
+        m_mainContour = new Contour(contourMat);
+        m_area = m_mainContour.getArea() / params.imageArea * 100;
+        m_fiducialId = result.getId();
+        m_shape = null;
+
+        // TODO implement skew? or just yeet
+
+
+        var tvec = new Mat(3, 1, CvType.CV_64FC1);
+        tvec.put(
+                0,
+                0,
+                new double[] {
+                    -bestPose.getTranslation().getY(),
+                    -bestPose.getTranslation().getZ(),
+                    bestPose.getTranslation().getX()
+                });
+        setCameraRelativeTvec(tvec);
+
+        // Opencv expects a 3d vector with norm = angle and direction = axis
+        Rotation3d newRot = new Rotation3d(bestPose.getRotation().getY(),bestPose.getRotation().getZ(), -bestPose.getRotation().getX());
+        var rvec = new Mat(3, 1, CvType.CV_64FC1);
+        MathUtils.rotationToOpencvRvec(newRot, rvec);
+        setCameraRelativeRvec(rvec);
```

Comment:
This smells a bit like a coordinate transform to me. Could we use the proper CoordinateSystem.convert call here, instead?

Reply:
Yeah